kedro/framework/project/__init__.py

```diff
@@ -184,6 +184,16 @@ def configure(self, pipelines_module: Optional[str] = None) -> None:
         self._is_data_loaded = False
         self._content = {}
 
+    def keys(self):
+        self._load_data()
+        return self._content.keys()
+    def values(self):
+        self._load_data()
+        return self._content.values()
+    def items(self):
+        self._load_data()
+        return self._content.items()
```

Comment:
```suggestion
    keys = _load_data_wrapper(operator.methodcaller("keys"))
    values = _load_data_wrapper(operator.methodcaller("values"))
    items = _load_data_wrapper(operator.methodcaller("items"))
```
and move it below to the `# Dict-like interface` section?

I think it's functionally equivalent to what you have (haven't tested it, just in my mind 🙈), just a bit more consistent with the existing code, and doesn't duplicate the `self._load_data()` and `self._content` access.

But yeah, makes sense to me overall to do this, and to provide a fuller dict-like interface. To be honest, it just took me a while to understand in my mind that the purpose of this PR is to essentially add missing parts of that dict-like interface.

Reply:
That's a better summary and I put that as the PR description instead.